controllers/machine_controller.go

```diff
@@ -491,6 +503,15 @@ func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func waitForHook(hookName string, annotations map[string]string) bool {
```

Comment:
This might come in handy for additional use cases, thoughts on moving this to `util/annotations` and changing the signature to something like:

```go
func HasAnnotationWithPrefix(prefix string, o metav1.Object) bool {}
```

Reply:
annotations.HasWithPrefix?

Reply:
I think annotations.HasWithPrefix going into utils so it can be imported by other libraries would be quite useful.